src/Sentry.Maui/Internal/ScreenshotAttachment.cs

```diff
@@ -0,0 +1,89 @@
+#if  __ANDROID__
+using Android.Util;
+#endif
+using Sentry.Extensibility;
+
+namespace Sentry.Maui.Internal;
+
+internal class ScreenshotAttachment : Attachment
+{
+    public ScreenshotAttachment(SentryMauiOptions options)
+        : this(
+            AttachmentType.Default,
+            new ScreenshotAttachmentContent(options),
+            "screenshot.jpg",
+            "image/jpeg")
+    {
+    }
+
+    private ScreenshotAttachment(
+        AttachmentType type,
+        IAttachmentContent content,
+        string fileName,
+        string? contentType)
+        : base(type, content, fileName, contentType)
+    {
+    }
+}
+
+internal class ScreenshotAttachmentContent : IAttachmentContent
+{
+    private readonly SentryMauiOptions _options;
+
+    //Used only in test to make it green if capture fails on Android
+    internal static bool CaptureFailed { get; set; }
+
+    public ScreenshotAttachmentContent(SentryMauiOptions options)
+    {
+        _options = options;
+    }
+
+    public Stream GetStream()
+    {
+        // Not including this on Windows specific build because on WinUI this can deadlock.
+#if !WINDOWS
+        var stream = MainThread.InvokeOnMainThreadAsync(async () =>
```

Comment:
Check if on UI thread. If yes, do as u did originally. Blocking (with commets to source showing it's not really blocking).
This guarantees there's no yelding back and possibly redrawing the screen.

Now if we're on teh background thread, and on iOS only (that's when it fails?) then you do this approach u did here. And we document that, saying that in this case it could be that the screen changed from the time of the error.

Reply:
Where do I document the iOS behavior?

Reply:
in Sentry docs, when you document the whole feature